### PR TITLE
[Core] Add `IsInside` method to `BoundingBox` class

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Add `IsInside` method to `BoundingBox` class. 

**🆕 Changelog**

- [Add `IsInside` method to `BoundingBox` class](https://github.com/KratosMultiphysics/Kratos/commit/b738a43006470a7d03f2c8b1e11c5e682527c64d)
- [Expose to python](https://github.com/KratosMultiphysics/Kratos/commit/8d9a925dd5e539c5ea136439d4771732bf893977)
- [Add test for `BoundingBox` `IsInside` method](https://github.com/KratosMultiphysics/Kratos/commit/cf37020264302b1d9b10759ebec7e648710c5454)

### CODE REVIEW

File: kratos/geometries/bounding_box.h [hunk 1]
```diff
@@ -167,6 +167,22 @@ class BoundingBox
...
 
+    /**
+     * @brief Checks if a point is inside the bounding box.
+     * @details This function checks if a given point is inside the bounding box defined by the
+     * minimum and maximum points. It returns true if the point is inside the bounding box,
+     * and false otherwise.
+     * @param rPoint The point to be checked.
+     * @return True if the point is inside the bounding box, false otherwise.
+     */
+    bool IsInside(const array_1d<double, 3>& rPoint) const
```
Comment (pooyan-dadvand) [inline kratos/geometries/bounding_box.h]: I would add a tolerance here (like the Geometry::IsInside)

Comment (loumalouomega) [reply kratos/geometries/bounding_box.h]: Done

**Comment by loumalouomega** [comment]
This is ready @pooyan-dadvand

**Comment by loumalouomega** [comment]
> This is ready @pooyan-dadvand

Hello @pooyan-dadvand

**Comment by loumalouomega** [comment]
> > This is ready @pooyan-dadvand
> 
> Hello @pooyan-dadvand

Or @rubenzorrilla

**Review by pooyan-dadvand — APPROVED** [review]

**Pull request merged**